src/prelude.rs

```diff
@@ -36,15 +36,15 @@
 //! ```
 //! use stm32f4xx_hal::prelude::*;
```

Comment:
if you want to make e-h 1.0 the default, shouldn't this then also be reflected in the prelude which should then export the e-h 1.0 traits instead of the e-h 0.2 traits? though this would be a breaking change for consumers relying on e-h 0.2 APIs which were re-exported here (as most of them changed between 0.2 and 1.0 or were removed altogether).

if you don't do the change now it'd at least be good to have an understanding for when this is planned to happen.